monitoring/src/minting-monitor.ts

```diff
@@ -178,4 +299,238 @@ export class MintingMonitor implements SystemEventMonitor {
+
+    return minters[index]
+  }
+
+  private async checkOrphanedMinting(
+    cache: ChainDataCache,
+    fromBlock: number,
+    toBlock: number
+  ): Promise<SystemEvent[]> {
+    const systemEvents: SystemEvent[] = []
+
+    systemEvents.push(
+      ...(await this.checkMintingNotRequested(fromBlock, toBlock))
+    )
+    systemEvents.push(
+      ...(await this.checkMintingNotFinalized(cache, fromBlock, toBlock))
+    )
+
+    return systemEvents
+  }
+
+  private async checkMintingNotRequested(
+    fromBlock: number,
+    toBlock: number
+  ): Promise<SystemEvent[]> {
+    const rewindBlock = (block: number, shift: number) =>
+      block - shift > 0 ? block - shift : 0
+
+    // We need to rewind the block window by the minting request timeout.
+    // This way, we are looking for past deposits whose time for creating
+    // the minting request was already elapsed.
+    const chainEvents = await this.bridge.getDepositRevealedEvents({
+      fromBlock: rewindBlock(fromBlock, mintingRequestTimeoutBlocks),
+      toBlock: rewindBlock(toBlock, mintingRequestTimeoutBlocks),
+    })
+
+    const mintingRequests = await Promise.allSettled(
+      chainEvents.map((ce) =>
+        OptimisticMinting.getOptimisticMintingRequest(
+          ce.fundingTxHash,
+          ce.fundingOutputIndex,
+          this.tbtcVault
+        )
+      )
+    )
+
+    return chainEvents
+      .filter((ce, index) => {
+        const request = mintingRequests[index]
+        const requestExists =
+          request.status === "fulfilled" && request.value.requestedAt !== 0
+        return !requestExists
+      })
+      .map(OptimisticMintingNotRequestedByAnyMinter)
+  }
+
+  private async checkMintingNotFinalized(
+    cache: ChainDataCache,
+    fromBlock: number,
+    toBlock: number
+  ): Promise<SystemEvent[]> {
+    const rewindBlock = (block: number, shift: number) =>
+      block - shift > 0 ? block - shift : 0
+
+    const finalizationTimeoutBlocks = mintingFinalizationTimeoutBlocks(
+      cache.optimisticMintingDelay
+    )
+
+    // We need to rewind the block window by the minting finalization timeout.
+    // This way, we are looking for past minting requests whose time for
+    // finalization was already elapsed.
+    const chainEvents =
+      await this.tbtcVault.getOptimisticMintingRequestedEvents({
+        fromBlock: rewindBlock(fromBlock, finalizationTimeoutBlocks),
+        toBlock: rewindBlock(toBlock, finalizationTimeoutBlocks),
+      })
+
+    const mintingRequests = await Promise.allSettled(
+      chainEvents.map((ce) =>
+        OptimisticMinting.getOptimisticMintingRequest(
+          ce.fundingTxHash,
+          ce.fundingOutputIndex,
+          this.tbtcVault
+        )
+      )
+    )
+
+    return chainEvents
+      .filter((ce, index) => {
+        const request = mintingRequests[index]
+        const requestExists =
+          request.status === "fulfilled" && request.value.finalizedAt !== 0
```

Comment:
IMO, this form is more readable. Otherwise, we'll need to use optional chaining (i.e. `request.value?. finalizedAt`) and check against `undefined` as well.